adapters/adocean/adocean.go

```diff
@@ -89,44 +96,58 @@ func (a *AdOceanAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adap
 		}
 	}
 
-	var httpRequests []*adapters.RequestData
 	var errors []error
-
+	requestsData := make([]*requestData, 0, len(request.Imp))
 	for _, auction := range request.Imp {
-		newHttpRequest, err := a.makeRequest(httpRequests, &auction, request, consentString)
+		err := a.addNewBid(&requestsData, &auction, request, consentString)
 		if err != nil {
 			errors = append(errors, err)
-		} else if newHttpRequest != nil {
-			httpRequests = append(httpRequests, newHttpRequest)
 		}
 	}
 
+	httpRequests := make([]*adapters.RequestData, 0, len(requestsData))
+	for _, requestData := range requestsData {
+		httpRequests = append(httpRequests, &adapters.RequestData{
+			Method:  "GET",
+			Uri:     requestData.Url,
+			Headers: *requestData.Headers,
+		})
+	}
+
 	return httpRequests, errors
 }
 
-func (a *AdOceanAdapter) makeRequest(existingRequests []*adapters.RequestData, imp *openrtb.Imp, request *openrtb.BidRequest, consentString string) (*adapters.RequestData, error) {
+func (a *AdOceanAdapter) addNewBid(
+	requestsData *[]*requestData,
```

Comment:
The usual go way of handling this is to return the `requestData` slice rather than passing a pointer in. I am not sure how go would handle the case where you would have to resize the slice larger, it would probably work, and should not be an issue with the current code. But I am not sure it will work as expected. Is there a reason you are not returning both `requestsData` and `error` instead of using this pointer magic?

Reply:
Well, It just seemed more logical to just pass a pointer to a slice if a function is going to change the slice size, instead of returning it, but I have no strong opinion on this matter and can change it so the slice will be return if you think that, it's better.

Reply:
It seems like your method should work, but every bit of go code I have ever seen that does something like that uses the return slice method. The go append builtin uses the return slice, which is as close to an official recommendation as you can get. So it should be safer to move to that method.

Reply:
It's changed.